test-framework/junit5-internal/src/main/java/io/quarkus/test/ExportUtil.java

```diff
@@ -35,4 +46,41 @@ static void exportToQuarkusDeploymentPath(JavaArchive archive) throws IOExceptio
         File exportFile = new File(exportDir, archive.getName());
         archive.as(ZipExporter.class).exportTo(exportFile);
     }
+
+    static void mergeCustomApplicationProperties(JavaArchive archive, Properties customApplicationProperties)
```

Comment:
I almost suggested you to do that, but decided against it as it felt too much like nitpicking...

Great minds think alike :D

Reply:
I guess this is a corner case and usually, we don't do it, but describing it for your consideration:

The resource archive may be a profiled file (`application-test.properties` or other), or even include both profiled and unprofiled. In the case of a profiled resource, that will always have priority over the unprofiled one.

Do we want to handle that case also? :)

Reply:
I have no idea :shrug:. 

In case of `QuarkusUnitTest` I think that we never use "profiled" files...